backend/main.cpp

```diff
@@ -173,11 +182,17 @@ void handle_request(const std::string &request, ip::tcp::socket &frontend_socket
                     matrices.result_matrix[i][j] = in_fpga_payload[pos++];
                 }
             }
+            const auto now = std::chrono::high_resolution_clock::now();
+            const std::chrono::duration<double> diff = now - timer;
+            std::cout << "PROFILER - Request to FPGA: " << diff.count() << " seconds." << std::endl;
         }
         else
         {
             // Multiply matrices 1 and 2 and store the result in the result matrix
             matrices.result_matrix = matrices.input_matrix_1 * matrices.input_matrix_2;
+            const auto now = std::chrono::high_resolution_clock::now();
+            const std::chrono::duration<double> diff = now - timer;
+            std::cout << "PROFILER - Request to digital: " << diff.count() << " seconds." << std::endl;
```

Comment:
I think it's okay to just have one "now" instead of splitting it up between FPGA/digital, since it should be pretty obvious which one is being used.